### DISCUSSION TITLE
How to get “today” or “now” in ICU

### DISCUSSION THREAD

**Discussion by Lucretiel** [unicode-org/icu4x#4575] [Q&A]
Is there a straightforward way to get "today's date" or "now" using `icu_calendar`? There don't seem to be any conversions from `Instant`, nor any functions in that crate resembling `now` or `today`, nor any constructors that don't have you bring all of your own datetime components. Is there something I'm missing? It seems like a shame to being *another* heavyweight timestamps crate, like `time` or `chrono`, just to do this initializer step for `icu`.

**Comment by sffc** [comment]
I agree that it would be useful to provide a well-lit path for formatting the current time in ICU4X. However, I'm not sure if doing it all internally in ICU4X is the right architecture. Some challenges:

1. The primary value proposition of ICU4X's primitives in `icu_calendar` is for converting between calendar systems and formatting localized output, something out of scope for `chrono` and `time`. It would be scope creep to include code for processing system times and time zones in ICU4X, which is a nontrivial amount of code to write and perpetually maintain.
2. `std::time` exposes a mechanism to get the current UTC timestamp, but it says nothing about time zone or local time, which means it is not enough on its own for extracting a `DateTime`. Some other crate is necessary in order to get information on the local time zone.

What _is_ in scope for ICU4X is the handling of time zone transitions and performing time zone calculations, since this is required by both ECMA-402 and Temporal and can be done as no-std. Therefore, I do expect ICU4X will get a function where the user provides the current timestamp and time zone and can get a `DateTime`. There could also be a convenience function gated on the `std` feature where the user provides only the time zone and ICU4X uses `std::time` to get the timestamp.

**Reply by savannidgerinel** [reply]
I think that's very reasonable. I'd be very happy with a function akin to as `DateTime::new_from_timestamp(std::time::SystemTime, [something timezone-ish])`, or `DateTime::new_from_chrono(chrono::DateTime<FixedOffset>)`.
